### PR TITLE
1.11.x - Adding api for user messages for webservices

### PR DESCRIPTION
**Pull request by hichamon306**
adding:
  functions to get user messages (sent and received).
 function to update user message status (delete, read)

### CODE REVIEW

**Comment by ywarnier** [comment]
@hichamon306 could you review the reports by codeclimate and flint an ensure these validate before we review your contribution, please?

**Comment by hichamon306** [comment]
@ywarnier  I have fixed all lint problems linked to my contribution

**Comment by ywarnier** [comment]
This looks very good. Thank you for your contribution and very sorry for the delay in including it.

File: main/inc/lib/message.lib.php [hunk 1]
```diff
@@ -2260,6 +2260,92 @@ public static function getMessagesFromLastReceivedMessage($userId, $lastId = 0)
...
 
+    /**
+     * Get the data of the last received messages for a user.
+     *
+     * @param int $userId The user id
+     * @param int $lastId The id of the last received message
+     *
+     * @return array
+     */
+    public static function getReceivedMessages($userId, $lastId = 0)
```
Comment (AngelFQC) [inline main/inc/lib/message.lib.php]: `MessageManager::getMessagesFromLastReceivedMessage` returns the last unread messages.

What's the difference with `MessageManager::getMessagesFromLastReceivedMessage`?

Comment (hichamon306) [reply main/inc/lib/message.lib.php]: Hi Angel,
MessageManager::getReceivedMessages this function return all received message (read & unread), I also add picture of sender.

Comment (AngelFQC) [reply main/inc/lib/message.lib.php]: :+1:

**Pull request merged**